### ISSUE TITLE
Invoking !python from MacVim does not use current virtualenv

### ISSUE THREAD

**Issue by kitmonisit** [Homebrew/homebrew#38253]
From a fresh install of Yosemite and Xcode, I installed Homebrew, and did the following:

    $ brew install python
    $ brew install --override-system-vim macvim
    # (after setting up virtualenvwrapper and creating virtual environment named FOOBAR):
    $ workon FOOBAR
    $ pip install numpy
    $ vim

From inside vim, I type `:!python` and press Enter. I get back to the shell and into a Python interpreter. So I type:

    >>> import numpy
    Traceback (most recent call last):
        File "<stdin>", line 1, in <module>
    ImportError: No module named numpy

However, if I simply type `:python import numpy` from within vim (no `!`), no problem.

The `!` is important to me because I have a shortcut along the lines of `!python %<cr>`. I think that this would ensure my script is running independent of the environment inside vim. This would also let me see the output of my script as though I ran it from the shell with `python somescript.py`,

I think that the problem here is that `!` commands from within vim should be as though I typed them into the shell prompt from which vim originated. In other words, it should be the same environment. I did not have this problem in my previous installations, though I cannot give a clear regression because I was a terribly inexperienced UNIX admin and my configuration was like spaghetti. It is only now that I attempted to clean up and start over.

For reference, I check `vim --version` and see that it is correctly linked against Homebrew Python in `/usr/local/Cellar/...`

## A different approach

I attempt to compile against the FOOBAR python:

    $ deactivate
    $ brew uninstall macvim
    $ workon FOOBAR
    $ brew install --override-system-vim -v macvim

From the verbose `./configure` output, I see the following lines:

    checking Python's install prefix... /Users/kit/UNIX/virtualenvs/FOOBAR/bin/..
    checking Python's execution prefix... /Users/kit/UNIX/virtualenvs/FOOBAR/bin/..
    checking Python's configuration directory... /usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/config

I still have the same problem as described above. Is there something in the formula that needs to be tweaked so that `!` commands from within vim would refer to the environment from which it started?

## Modifying PYTHONPATH

In my `.zshrc`, I have the following lines:

    vim() {
        PYTHONPATH=$VIRTUAL_ENV/lib/python2.7/site-packages \
            $LOCAL/bin/vim "$@"
    }

However, I think that one of Homebrew's objectives is to let users install software without further hacking into configuration files as much as possible. So it would be nice if I didn't need to add those lines in my `.zshrc`. It is also `virtualenv`'s objective that I do not have to modify PYTHONPATH to get to the relevant packages of my current project.

Link to the gist-logs here: https://gist.github.com/kitmonisit/a36152e0f0f712ad02cb

**Comment by kitmonisit** [comment]
#### TL;DR
I found a workaround:

    $ sudo ugo-x /usr/libexec/path_helper

It appears this problem has been reported in Stack Overflow. I'll trace how I found it and add comments if the link name does not tell much:

- [Set python virtualenv in vim](http://stackoverflow.com/q/3881534/366309)
- One of the comments in [one of the answers](http://stackoverflow.com/a/16497692/366309) led me to an answer in another question:
- [A known problem introduced by Apple in OS X Leopard](http://stackoverflow.com/a/12146694/366309)
- Which eventually led me to [dotphiles/dotzsh README](https://github.com/dotphiles/dotzsh#mac-os-x)

I quote the explanation:

> ### Mac OS X

> If you have administrator privileges, you must fix an Apple-introduced problem
> in Mac OS X 10.5 Leopard by executing the following command, or BASH and Zsh
> will have the wrong `PATH` when executed non-interactively.
> 
>     sudo chmod ugo-x /usr/libexec/path_helper
> 
> `path_helper` is intended to make it easier for installers to add new paths to
> the environment without having to edit shell configuration files by adding
> a file with a path to the */etc/paths.d* directory.
> 
> Unfortunately, `path_helper` always reads paths from */etc/paths* set by Apple
> then paths from */etc/paths.d* set by third party installers, and lastly paths
> from the `PATH` environment variable set by the parent process, which
> ultimately is set by the user with `export PATH=...` Thus, it reorders path
> priorities, and user */bin* directories meant to override system */bin*
> directories end up at the tail of the array.

**Comment by kitmonisit** [comment]
I will now close this issue as this is neither a Homebrew nor MacVim problem.

**Comment by kitmonisit** [comment]
I found a more precise solution. Instead of doing

    $ sudo ugo-x /usr/libexec/path_helper

Do the following instead: Edit `/etc/zshenv`. Its original contents look like this:

    # system-wide environment settings for zsh(1)
    if [ -x /usr/libexec/path_helper ]; then
            eval `/usr/libexec/path_helper -s`
    fi

Make it look like this:

    # system-wide environment settings for zsh(1)
    if [ -x /usr/libexec/path_helper ]; then
    # The following if block is found in
    # zsh.sourceforge.net/Guide/zshguide02.html
        if [[ ! -o rcs ]]; then
            eval `/usr/libexec/path_helper -s`
        fi
    fi

Inferred from [this portion of the zsh Guide](http://zsh.sourceforge.net/Guide/zshguide02.html), section __2.2 All the startup files__
